controllers/space/space_controller_test.go

```diff
@@ -833,6 +834,77 @@ func TestUpdateSpaceTier(t *testing.T) {
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"),
+			spacetest.WithFinalizer(),
+			spacetest.WithCondition(spacetest.Ready()))
+		nsTmplSet1 := nstemplatetsettest.NewNSTemplateSet(s1.Name,
+			nstemplatetsettest.WithReferencesFor(olderBasicTier), // NSTemplateSet has references to old basic tier
+			nstemplatetsettest.WithReadyCondition())
+
+		t.Run("when is postponed by two seconds", func(t *testing.T) {
+			hostClient := test.NewFakeClient(t, s1, basicTier)
+			member1Client := test.NewFakeClient(t, nsTmplSet1)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			ctrl := newReconciler(hostClient, member1)
+			ctrl.LastExecutedUpdate = time.Now()
+
+			// when reconciling space `s1`
+			res, err := ctrl.Reconcile(context.TODO(), requestFor(s1))
+
+			// then
+			require.NoError(t, err)
+			assert.True(t, res.Requeue)
+			assert.LessOrEqual(t, res.RequeueAfter, 2*time.Second) // wait 2s for NSTemplateSet update to begin
+			assert.LessOrEqual(t, time.Until(ctrl.NextScheduledUpdate), 2*time.Second)
+			// check that the NSTemplateSet is not being updated
+			spacetest.AssertThatSpace(t, test.HostOperatorNs, s1.Name, hostClient).
+				Exists().
+				HasTier(basicTier.Name).
+				HasConditions(spacetest.Ready()).
+				HasMatchingTierLabelForTier(olderBasicTier)
+			nsTmplSet1 = nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, nsTmplSet1.Name, member1.Client).
+				Exists().
+				Get()
+			require.True(t, tierutil.TierHashMatches(olderBasicTier, nsTmplSet1.Spec))
+		})
+
+		t.Run("when is postponed by two seconds from the NextScheduledUpdate", func(t *testing.T) {
+			hostClient := test.NewFakeClient(t, s1, basicTier)
+			member1Client := test.NewFakeClient(t, nsTmplSet1)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			ctrl := newReconciler(hostClient, member1)
+			ctrl.NextScheduledUpdate = time.Now().Add(1 * time.Minute)
+			ctrl.LastExecutedUpdate = time.Now()
+
+			// when reconciling space `s1`
+			res, err := ctrl.Reconcile(context.TODO(), requestFor(s1))
+
+			// then
+			require.NoError(t, err)
+			assert.True(t, res.Requeue)
+			assert.LessOrEqual(t, res.RequeueAfter, time.Minute+(2*time.Second)) // wait 2s for NSTemplateSet update to begin
+			assert.LessOrEqual(t, time.Until(ctrl.NextScheduledUpdate), time.Minute+(2*time.Second))
+			// check that the NSTemplateSet is not being updated
+			spacetest.AssertThatSpace(t, test.HostOperatorNs, s1.Name, hostClient).
+				Exists().
+				HasTier(basicTier.Name).
+				HasConditions(spacetest.Ready()).
+				HasMatchingTierLabelForTier(olderBasicTier)
+			nsTmplSet1 = nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, nsTmplSet1.Name, member1.Client).
+				Exists().
+				Get()
+			require.True(t, tierutil.TierHashMatches(olderBasicTier, nsTmplSet1.Spec))
+		})
+
+	})
```

Comment:
could we also verify the case when the Space's condition is not ready (eg. failed or updating) so it would ignore the postpone logic?

Reply:
see 177f220 and cf44446